modules/tinymce/src/core/test/ts/browser/textpatterns/FindBlockPatternTest.ts

```diff
@@ -0,0 +1,207 @@
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      text_patterns: [
+        { start: '*', end: '*', format: 'italic' },
+        { start: '**', end: '**', format: 'bold' },
+        { start: '#', format: 'h1' },
+        { start: '##', format: 'h2' },
+        { start: '###', format: 'h3' },
+        { start: '' }
+      ],
+      text_patterns_lookup: (_ctx) => {
+        return [
+          { start: '####', format: 'h4' },
+          { start: 'TBA', cmd: 'mceInsertContent', value: 'To be announced' },
+          { start: '###', cmd: 'mceInsertContent', value: 'h3 heading' }
+        ];
+      },
+      base_url: '/project/tinymce/js/tinymce'
+    }, []);
+
+    const getPatternSet = getPatternSetFor(hook);
+
+    it('TINY-8778: Match a heading format which is extending the existing text patterns', () => {
+      const editor = hook.editor();
+      editor.setContent('<p>#### wow</p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 4);
+      const matches = BlockPattern.findPatterns(editor, getPatternSet());
+      assert.deepEqual(
+        matches,
+        [
+          {
+            pattern: {
+              type: 'block-format',
+              start: '####',
+              format: 'h4'
+            },
+            range: {
+              start: [ 0, 0 ],
+              end: [ 0, 0 ]
+            }
+          }
+        ],
+        'Checking block pattern matches'
+      );
+    });
+
+    it('TINY-8778: Match a command pattern', () => {
+      const editor = hook.editor();
+      editor.setContent('<p><b>TBA Bold heading</b></p>');
+      TinySelections.setCursor(editor, [ 0, 0, 0 ], 3);
+      const matches = BlockPattern.findPatterns(editor, getPatternSet());
+      assert.deepEqual(
+        matches,
+        [
+          {
+            pattern: {
+              type: 'block-command',
+              start: 'TBA',
+              cmd: 'mceInsertContent',
+              value: 'To be announced'
+            },
+            range: {
+              start: [ 0, 0 ],
+              end: [ 0, 0 ]
+            }
+          }
+        ],
+        'Checking block pattern matches'
+      );
+    });
+
+    it('TINY-8778: Match a heading where its block tag matches forced_root_block but its parent block does not', () => {
+      const editor = hook.editor();
+      editor.setContent('<div><p>#### New heading type</p></div>');
+      TinySelections.setCursor(editor, [ 0, 0, 0 ], 4);
+      const patterns = getPatternSet();
+      const matches = BlockPattern.findPatterns(editor, patterns);
+      assert.deepEqual(matches, [
+        {
+          pattern: {
+            type: 'block-format',
+            start: '####',
+            format: 'h4'
+          },
+          range: {
+            start: [ 0, 0, 0 ],
+            end: [ 0, 0, 0 ]
+          }
+        }
+      ],
+      'Checking block pattern matches'
+      );
+    });
+
+    it('TINY-8778: Does not trigger a match if the block tag does not match forced_root_block', () => {
```

Comment:
Likewise here, this isn't something that's expected so please either remove it or add a comment referencing TINY-4303.